### PR TITLE
Simplify authority check #45

### PR DESCRIPTION
**Pull request by s-medvedev**

### CODE REVIEW

File: evm_loader/program/src/transaction.rs [hunk 1]
```diff
@@ -120,24 +120,16 @@ impl rlp::Decodable for UnsignedTransaction {
...
     let signature = secp256k1::Signature::parse_slice(&signature[0..64])?;
 
     let public_key = secp256k1::recover(&message, &signature, &recovery_id)?;
-    if secp256k1::verify(&message, &signature, &public_key) {
-        Ok(())
-    } else {
-        Err(secp256k1::Error::InvalidSignature)
-    }
+    let address = keccak256_digest(&public_key.serialize()[1..]);
+    let address = H160::from_slice(&address[12..32]);
```
Comment (anton-lisanin) [inline evm_loader/program/src/transaction.rs:132]: You are using 2 different ways to compute ether address from the public key. Chose one of them and stick to it.

There and https://github.com/neonlabsorg/neon-evm/pull/46/files#diff-77d828dbc5822cedb23202df3c9fd48d8be70932389c3199d0b2f888387a0cffR112

File: evm_loader/program/src/entrypoint.rs [hunk 2]
```diff
@@ -731,51 +653,32 @@ fn invoke_on_return<'a>(
...
-                    return Err(ProgramError::InvalidArgument);
-                }
-            }
-        } else {
-            return Err(ProgramError::InvalidArgument);
-        }
+    if SolanaBackend::<ProgramAccountStorage>::chain_id() != *chain_id {
+        debug_print!("Invalid chain_id: actual {}, expected {}",
+                &chain_id.to_string(),
+                &SolanaBackend::<ProgramAccountStorage>::chain_id().to_string());
```
Comment (anton-lisanin) [inline evm_loader/program/src/entrypoint.rs:680]: `&chain_id.to_string()` can be simplified to `chain_id`.
Also check other debug_print macro

Comment (s-medvedev) [reply evm_loader/program/src/entrypoint.rs]: Fixed

File: evm_loader/program/src/transaction.rs (hunk 1 above)
Comment (s-medvedev) [reply evm_loader/program/src/transaction.rs]: Fixed

**Review by anton-lisanin — APPROVED** [review]

**Pull request merged**